lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskRuntime.cs

```diff
@@ -31,58 +31,41 @@
 
 namespace Org.Apache.REEF.Common.Runtime.Evaluator.Task
 {
-    public class TaskRuntime : IObserver<ICloseEvent>, IObserver<ISuspendEvent>, IObserver<IDriverMessage>
+    internal sealed class TaskRuntime : IObserver<ICloseEvent>, IObserver<ISuspendEvent>, IObserver<IDriverMessage>
     {
         private static readonly Logger LOGGER = Logger.GetLogger(typeof(TaskRuntime));
-        
-        private readonly ITask _task;
 
         private readonly IInjector _injector;
 
-        // The memento given by the task configuration
-        private readonly Optional<byte[]> _memento;
-
-        private readonly HeartBeatManager _heartBeatManager;
-
         private readonly TaskStatus _currentStatus;
 
-        private readonly INameClient _nameClient;
-
         private readonly Lazy<IDriverConnectionMessageHandler> _driverConnectionMessageHandler;
 
-        public TaskRuntime(IInjector taskInjector, string contextId, string taskId, HeartBeatManager heartBeatManager, string memento = null)
+        private readonly Lazy<IDriverMessageHandler> _driverMessageHandler;
+
+        public TaskRuntime(IInjector taskInjector, string contextId, string taskId, HeartBeatManager heartBeatManager)
         {
             _injector = taskInjector;
-            _heartBeatManager = heartBeatManager;
 
-            Optional<ISet<ITaskMessageSource>> messageSources = Optional<ISet<ITaskMessageSource>>.Empty();
-            try
-            {
-                _task = _injector.GetInstance<ITask>();
```

Comment:
Good point, I haven't considered re-entrancy here. That said, I don't think saving it as a class variable is the way to go. I'll use a flag instead.